tests/standalone_activity_test.go

```diff
@@ -1283,6 +1285,322 @@ func (s *standaloneActivityTestSuite) TestGetActivityExecutionOutcome_InvalidArg
 
 // TODO(dan): add tests that DescribeActivityExecution can wait for deletion, termination, cancellation etc
 
+func (s *standaloneActivityTestSuite) TestListActivityExecutions() {
+	t := s.T()
+	ctx, cancel := context.WithTimeout(t.Context(), 30*time.Second)
+	defer cancel()
+
+	activityID := s.tv.ActivityID()
+	activityType := s.tv.ActivityType().GetName()
+	startResp := s.startAndValidateActivity(ctx, t, activityID, s.tv.TaskQueue().GetName())
+	runID := startResp.RunId
+
+	verifyListQuery := func(t *testing.T, query string) {
+		t.Helper()
+		var resp *workflowservice.ListActivityExecutionsResponse
+		s.Eventually(
+			func() bool {
+				var err error
+				resp, err = s.FrontendClient().ListActivityExecutions(ctx, &workflowservice.ListActivityExecutionsRequest{
+					Namespace: s.Namespace().String(),
+					PageSize:  10,
+					Query:     query,
+				})
+				return err == nil && len(resp.GetExecutions()) >= 1
+			},
+			testcore.WaitForESToSettle,
+			100*time.Millisecond,
+		)
+		require.Len(t, resp.GetExecutions(), 1, "expected exactly 1 result for query: %s", query)
+		exec := resp.GetExecutions()[0]
+		s.Equal(activityID, exec.GetActivityId())
+		s.Equal(runID, exec.GetRunId())
+		s.Equal(activityType, exec.GetActivityType().GetName())
+		s.Equal(enumspb.ACTIVITY_EXECUTION_STATUS_RUNNING, exec.GetStatus())
+		s.NotNil(exec.GetScheduleTime())
+	}
+
+	t.Run("QueryByActivityId", func(t *testing.T) {
+		verifyListQuery(t, fmt.Sprintf("ActivityId = '%s'", activityID))
+	})
+
+	t.Run("QueryByActivityType", func(t *testing.T) {
+		verifyListQuery(t, fmt.Sprintf("ActivityType = '%s'", activityType))
+	})
+
+	t.Run("QueryByActivityStatus", func(t *testing.T) {
+		verifyListQuery(t, fmt.Sprintf("ActivityStatus = 'Scheduled' AND ActivityType = '%s'", activityType))
+	})
+
+	t.Run("QueryByTaskQueue", func(t *testing.T) {
+		verifyListQuery(t, fmt.Sprintf("ActivityTaskQueue = '%s' AND ActivityType = '%s'", s.tv.TaskQueue().GetName(), activityType))
+	})
+
+	t.Run("QueryByMultipleFields", func(t *testing.T) {
+		verifyListQuery(t, fmt.Sprintf("ActivityId = '%s' AND ActivityType = '%s'", activityID, activityType))
+	})
+
+	t.Run("QueryByCustomSearchAttribute", func(t *testing.T) {
+		customSAName := "ActivityCustomKeyword"
+		customSAValue := "custom-sa-test-value"
+
+		_, err := s.OperatorClient().AddSearchAttributes(ctx, &operatorservice.AddSearchAttributesRequest{
```

Comment:
I think there's a set of custom search attributes already defined for all functional tests.